### ISSUE TITLE
ERROR  getting API info for Juju 2 devel on non standard bridge

### ISSUE THREAD

**Issue by niccolox** [juju/juju#5439]
still having problems with Juju 2 (devel) and LXD on Xenial

I am using lxd on br0 happily with lxc via command line and also now the new LXD Ansible cloud module (still a PR I think)

Juju 2 still doesnt work for lxd

always getting this error "

> getting API info

and in my logs 

> LXD isn't socket activated

is this https://bugs.launchpad.net/juju-core/+bug/1571593

or  https://github.com/juju/juju/issues/4652

Juju has created all the files as required

```
lxd       
WARN[05-20|21:28:52] Per-container AppArmor profiles are disabled because the mac_admin capability is missing. 
WARN[05-20|21:28:52] CGroup memory swap accounting is disabled, swap limits will be ignored. 
error: remove /var/lib/lxd/devlxd/sock: permission denied
```

```
tree /home/devekko/.local/share/juju/                         
/home/devekko/.local/share/juju/
├── accounts.yaml
├── bootstrap-config.yaml
├── controllers.yaml
├── current-controller
├── models.yaml
└── ssh
    ├── juju_id_rsa
    └── juju_id_rsa.pub

```
1 directory, 7 files

my host networking

```
auto lo
iface lo inet loopback

iface enp2s0f0 inet manual

auto br0
iface br0 inet dhcp

auto br0:0
iface br0:0 inet static
    address 92.93.187.125/28

dns-nameservers 8.8.8.8 72.72.72.72
dns-search      demo.net
```

```
sudo -E lxd shutdown             
demo@blahblah:~|⇒  sudo -E lxd --group sudo --debug   
INFO[05-20|19:45:41] LXD is starting in normal mode           path=/var/lib/lxd
WARN[05-20|19:45:41] CGroup memory swap accounting is disabled, swap limits will be ignored. 
INFO[05-20|19:45:41] Default uid/gid map: 
INFO[05-20|19:45:41]  - u 0 165536 65536 
INFO[05-20|19:45:41]  - g 0 165536 65536 
INFO[05-20|19:45:41] Init                                     driver=storage/zfs
INFO[05-20|19:45:41] Looking for existing certificates        cert=/var/lib/lxd/server.crt key=/var/lib/lxd/server.key
INFO[05-20|19:45:41] LXD isn't socket activated 
DBUG[05-20|19:45:41] Raw response: {"type":"sync","status":"Success","status_code":200,"metadata":{"api_extensions":[],"api_status":"stable","api_version":"1.0","auth":"trusted","config":{"core.https_address":"[::]","storage.zfs_pool_name":"lxd"},"environment":{"addresses":["REDACTED:8443","REDACTED:8443","[2601:645:2016a:8092]:8443","192.168.121.1:8443","192.168.99.1:8443"],"architectures":["x86_64","i686"],"certificate":"-----BEGIN CERTIFICATE-----\n","driver":"lxc","driver_version":"2.0.0","kernel":"Linux","kernel_architecture":"x86_64","kernel_version":"4.4.0-22-generic","server":"lxd","server_pid":21088,"server_version":"2.0.0","storage":"zfs","storage_version":"5"},"public":false}}
 
error: LXD is already running.
```

and

```
sudo systemctl status lxd 
● lxd.service - LXD - main daemon
   Loaded: loaded (/lib/systemd/system/lxd.service; indirect; vendor preset: enabled)
   Active: active (running) since Fri 2016-05-20 19:45:41 PDT; 8min ago
     Docs: man:lxd(1)
  Process: 21092 ExecStartPost=/usr/bin/lxd waitready --timeout=600 (code=exited, status=0/SU
  Process: 21089 ExecStartPost=/usr/lib/lxd/profile-config (code=exited, status=0/SUCCESS)
  Process: 21077 ExecStartPre=/usr/lib/x86_64-linux-gnu/lxc/lxc-apparmor-load (code=exited, s
 Main PID: 21088 (lxd)
    Tasks: 51
   Memory: 45.6M
      CPU: 1.967s
   CGroup: /system.slice/lxd.service
           ├─ 2484 [lxc monitor] /var/lib/lxd/containers ws02
           ├─21088 /usr/bin/lxd --group lxd --logfile=/var/log/lxd/lxd.log
           ├─21137 [lxc monitor] /var/lib/lxd/containers atlassian-xenial
           ├─21246 [lxc monitor] /var/lib/lxd/containers bigbluebutton
           ├─22303 [lxc monitor] /var/lib/lxd/containers centos7
           ├─24101 [lxc monitor] /var/lib/lxd/containers edx
           ├─25194 [lxc monitor] /var/lib/lxd/containers gitlab-xenial
  
May 20 19:45:41 mendix systemd[1]: Starting LXD - main daemon...
May 20 19:45:41 mendix lxd[21088]: t=2016-05-20T19:45:41-0700 lvl=warn msg="CGroup memory
May 20 19:45:41 mendix systemd[1]: Started LXD - main daemon.
```

/etc/default/lxd-bridge

```
# WARNING: This file is generated by a debconf template!
# It is recommended to update it by using "dpkg-reconfigure -p medium lxd"

# Whether to setup a new bridge or use an existing one
USE_LXD_BRIDGE="false"

# Bridge name
# This is still used even if USE_LXD_BRIDGE is set to false
# set to an empty value to fully disable
LXD_BRIDGE="br0"

# Update the "default" LXD profile
UPDATE_PROFILE="true"

# Path to an extra dnsmasq configuration file
LXD_CONFILE=""

# DNS domain for the bridge
LXD_DOMAIN="lxd"

# IPv4
## IPv4 address (e.g. 10.0.8.1)
LXD_IPV4_ADDR="10.25.57.1"

## IPv4 netmask (e.g. 255.255.255.0)
LXD_IPV4_NETMASK="255.255.255.0"

## IPv4 network (e.g. 10.0.8.0/24)
LXD_IPV4_NETWORK="10.25.57.1/24"

## IPv4 DHCP range (e.g. 10.0.8.2,10.0.8.254)
LXD_IPV4_DHCP_RANGE="10.25.57.2,10.25.57.254"

## IPv4 DHCP number of hosts (e.g. 250)
LXD_IPV4_DHCP_MAX="252"

## NAT IPv4 traffic
LXD_IPV4_NAT="true"

# IPv6
## IPv6 address (e.g. 2001:470:b368:4242::1)
LXD_IPV6_ADDR="fd97:b6ef:3cef:85e::1"

## IPv6 CIDR mask (e.g. 64)
LXD_IPV6_MASK="64"

## IPv6 network (e.g. 2001:470:b368:4242::/64)
LXD_IPV6_NETWORK="fd97:b6ef:3cef:85e::1/64"


```

lxc profile edit default

```
name: default
config: {}
description: Default LXD profile
devices:
  eth0:
    name: eth0
    nictype: bridged
    parent: br0
    type: nic

```
```

netstat -i   
Kernel Interface table
Iface   MTU Met   RX-OK RX-ERR RX-DRP RX-OVR    TX-OK TX-ERR TX-DRP TX-OVR Flg
br0        1500 0   1261551      0      0 0        912703      0      0      0 BMRU
br0:0      1500 0       - no statistics available -                        BMRU
enp2s0f0   1500 0   2163018      0      0 0        818739      0      0      0 BMRU
lo        65536 0     39503      0      0 0         39503      0      0      0 LRU
veth0B0M97  1500 0       243      0      0 0         11828      0      0      0 BMRU
veth4NNUM6  1500 0       275      0      0 0         11871      0      0      0 BMRU
vethB18PG7  1500 0       270      0      0 0         11879      0      0      0 BMRU
vethF1PK5J  1500 0       242      0      0 0         11833      0      0      0 BMRU
vethG7H39F  1500 0       244      0      0 0         11861      0      0      0 BMRU
vethJCW1XJ  1500 0       239      0      0 0         12048      0      0      0 BMRU
vethN0UOEH  1500 0     58746      0      0 0         56968      0      0      0 BMRU
vethN536NO  1500 0       618      0      0 0         12549      0      0      0 BMRU
vethOESAB7  1500 0      1327      0      0 0         12875      0      0      0 BMRU
vethRW3IRE  1500 0       243      0      0 0         11852      0      0      0 BMRU
vethUHX520  1500 0       681      0      0 0         12318      0      0      0 BMRU
vethY3DOQC  1500 0       246      0      0 0         12284      0      0      0 BMRU
vmnet1     1500 0         0      0      0 0           134      0      0      0 BMRU
vmnet8     1500 0         0      0      0 0           132      0      0      0 BMRU
```

```
br0       Link encap:Ethernet  HWaddr 00:25:90:6a:80:92  
          inet addr:10.1.10.98  Bcast:10.1.10.255  Mask:255.255.255.0
          inet6 addr: fe80::225:90ff:fe6a:8092/64 Scope:Link
          inet6 addr: 2601:645:201:9000:225:90ff:fe6a:8092/64 Scope:Global
          inet6 addr: 2601:645:201:9000:a810:9703:9ec1:5888/64 Scope:Global
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:1283031 errors:0 dropped:0 overruns:0 frame:0
          TX packets:928036 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:3140969903 (3.1 GB)  TX bytes:315332522 (315.3 MB)

br0:0     Link encap:Ethernet  HWaddr 00:25:90:6a:80:92  
          inet addr:REDACTED  Bcast:REDACTED.127  Mask:255.255.255.240
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1

enp2s0f0  Link encap:Ethernet  HWaddr 00:25:90:6a:80:92  
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:2204451 errors:0 dropped:0 overruns:0 frame:0
          TX packets:833559 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:3210565176 (3.2 GB)  TX bytes:82074558 (82.0 MB)
          Memory:df420000-df43ffff 

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:39919 errors:0 dropped:0 overruns:0 frame:0
          TX packets:39919 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1 
          RX bytes:12561545 (12.5 MB)  TX bytes:12561545 (12.5 MB)

veth0B0M97 Link encap:Ethernet  HWaddr fe:2e:15:d8:52:18  
          inet6 addr: fe80::fc2e:15ff:fed8:5218/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:248 errors:0 dropped:0 overruns:0 frame:0
          TX packets:12067 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:13864 (13.8 KB)  TX bytes:1007951 (1.0 MB)
```

**Comment by niccolox** [comment]
now after trying various I am back to 

```
juju bootstrap lxd-tests3 lxd                    
ERROR cannot find network interface "lxdbr0": route ip+net: no such network interface
ERROR invalid config: route ip+net: no such network interface

```

tried this

```
sudo lxd --debug      
INFO[05-21|13:51:47] LXD is starting in normal mode           path=/var/lib/lxd
WARN[05-21|13:51:47] CGroup memory swap accounting is disabled, swap limits will be ignored. 
INFO[05-21|13:51:47] Default uid/gid map: 
INFO[05-21|13:51:47]  - u 0 165536 65536 
INFO[05-21|13:51:47]  - g 0 165536 65536 
INFO[05-21|13:51:47] Init                                     driver=storage/zfs
INFO[05-21|13:51:47] Looking for existing certificates        cert=/var/lib/lxd/server.crt key=/var/lib/lxd/server.key
INFO[05-21|13:51:47] LXD isn't socket activated 
DBUG[05-21|13:51:47] Detected stale unix socket, deleting 
DBUG[05-21|13:51:47] Pruning expired images 
DBUG[05-21|13:51:47] Updating images 
INFO[05-21|13:51:47] REST API daemon: 
INFO[05-21|13:51:47]  - binding Unix socket                   socket=/var/lib/lxd/unix.socket
INFO[05-21|13:51:47]  - binding TCP socket                    socket=[::]:8443

```

and 

lxc list                                                
+------------------------+---------+--------------------------------+---------------------------------------------+------------+-----------+
|          NAME          |  STATE  |              IPV4              |                    IPV6                     |    TYPE    | SNAPSHOTS |
+------------------------+---------+--------------------------------+---------------------------------------------+------------+-----------+
| atlassian-xenial       | STOPPED |                                |                                             | PERSISTENT | 0         |
+------------------------+---------+--------------------------------+---------------------------------------------+------------+-----------+
| bigbluebutton          | STOPPED |                                |                                             | PERSISTENT | 0         |
+------------------------+---------+--------------------------------+---------------------------------------------+------------+-----------+
| centos7                | STOPPED |                                |                                             | PERSISTENT | 0         |
+------------------------+---------+--------------------------------+---------------------------------------------+------------+-----------+
| edx       | STOPPED |                                |                                             | PERSISTENT | 0         |
+------------------------+---------+--------------------------------+---------------------------------------------+------------+-----------+
| gitlab-xenial          | RUNNING | REDACTED (eth0)           | 2601:645:201:9000:216:3eff:fe83:489d (eth0) | PERSISTENT | 0         |
|                        |         | 10.1.10.116 (eth0)             |                                             |            |           |
|                        |         | 10.0.3.1 (lxcbr0)              |                                             |            |           |
+------------------------+---------+--------------------------------+---------------------------------------------+------------+------

**Comment by niccolox** [comment]
and tested again with Ansible 2.2. from source and the Ansible LXD Container cloud module

play

```
---
- hosts: localhost
  tasks:
  - lxd_container:
      name: nettest2
      state: started
      config:
        source:
          type: image
          mode: pull
          server: https://images.linuxcontainers.org
          protocol: lxd
          alias: "ubuntu/xenial/amd64"
        profiles: ["default"]


```

lxc list
`
| nettest2               | RUNNING | 10.1.10.33 (eth0)              | 2601:645:201:9000:216:3eff:fe34:594a (eth0) | PERSISTENT | 0         |`

**Comment by niccolox** [comment]
looking at the docs I am wondering now if its even possible to use another bridge other than lxdbr0 with juju 2 ? 

I dont want to use lxdbr0, I want a custom bridge br0 and it works fine for LXD lxc cli and Ansible cloud module

```
sudo dpkg-reconfigure -p medium lxd
!!! Note: During the configuration dialog, the bridge must be 'lxdbr0'. You can accept all other prompts although the last question (IPv6) is probably not needed.

```
https://github.com/juju/docs/blob/master/src/en/getting-started.md

**Comment by niccolox** [comment]
am wondering if its related to https://github.com/lxc/lxd/issues/2040

**Comment by cherylj** [comment]
Relevant bug:  https://bugs.launchpad.net/juju-core/+bug/1575676

**Comment by dimitern** [comment]
Closing this as it's tracked in LP and there's a fix for it, soon to land - #5871
